### PR TITLE
delete unpublished courses originally imported from ocw-to-hugo

### PR DESCRIPTION
**Pull request by gumaerc**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-studio/issues/522

#### What's this PR do?
When courses are unpublished in the current Plone based OCW system, this information propagates through the system and eventually `ocw-to-hugo` will see this and decide not to output data for that course.  In `ocw-studio`, `import_ocw_course_sites` sources its data from an S3 bucket filled with `ocw-to-hugo` output.  Since `ocw-studio` is not set up to handle this, when courses are unpublished and subsequently removed from the bucket, the `Website` objects linger in the database.  This PR adds functionality that by default will remove sites for courses that are no longer present in the `ocw-to-hugo` output bucket.  You might also notice that the `Website` name property is now being sourced from the folder name and not from `course_id` in the metadata, as this metadata property will soon be removed.

#### How should this be manually tested?
Since `import_ocw_course_sites` is only set up to use S3, a fake S3 bucket can be mocked in unit testing (and it is) but when it comes to running a manual test it's a bit more tricky.

 - Spin up `ocw-studio` locally and make sure you have S3 credentials configured
 - Make sure you have the `ocw-www` and `ocw-course` website starters from https://github.com/mitodl/ocw-hugo-projects loaded into your local instance and create an empty `ocw-www` site named `ocw-www`
 - Run `docker-compose run web ./manage.py import_ocw_course_sites -b ocw-to-hugo-output-qa --filter 5-310`
 - Drop into a Django shell with `docker-compose run web ./manage.py shell`
 - Run the following:
```
from websites.models import Website
Website.objects.all().count()
```
- Verify that the number of websites returned is 3
- Insert a line into `ocw_import/api.py` after line 360 to spoof one of the courses being removed from S3:
`paths = ["15-310-managerial-psychology-laboratory-spring-2003/data/course.json"]`
- Rebuild your docker containers
- Re-run the same `import_ocw_course_sites` command from above
- Re-run the same check in a Django shell and verify that the amount of `Website` objects is now 2

### CODE REVIEW

File: ocw_import/api.py [hunk 1]
```diff
@@ -348,3 +348,24 @@ def generate_topics_dict(course_paths, bucket_name):
...
+        filter_str (str): (Optional) If specified, filter courses to remove
+    """
+    if not paths:
+        return
+    course_ids = list(map((lambda key: key.replace("/data/course.json", "", 1)), paths))
+    unpublished_courses = Website.objects.filter(
+        source=WEBSITE_SOURCE_OCW_IMPORT
+    ).exclude(name__in=course_ids)
+    if filter_str:
+        unpublished_courses = unpublished_courses.filter(name__contains=filter_str)
```
Comment (mbertrand) [inline ocw_import/api.py:368]: Can you add a unit test to cover this line?

**Review by mbertrand — CHANGES_REQUESTED** [review]
Functionality works great

**Comment by gumaerc** [comment]
@mbertrand all set

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**